File: src/model/surveymodel.ts
import  {Survey } from "@prisma/client"

export interface SurveyResponse {
    id: number
    title: string
    start_date: string
    end_date: string
}

// Fungsi untuk mengonversi Survey menjadi SurveyResponse
export function toSurveyResponseList(prismaSurvey: Survey[]): SurveyResponse[] {
    const result = prismaSurvey.map((survey) => {
        return {
            id: survey.id,
            title: survey.title,
            start_date: survey.start_date,
            end_date: survey.end_date,
        }
    })

    return result
}

export function toSurveyResponse(prismaSurvey: Survey): SurveyResponse {
    return {
        id: prismaSurvey.id,
        title: prismaSurvey.title,
        start_date: prismaSurvey.start_date,
        end_date: prismaSurvey.end_date,
    }
}

export interface SurveyCreateRequest {
    title: string
    start_date: string
    end_date: string
}

File: models/SurveyModel.kt
package com.example.alp_vp.models

data class GetAllSurveyResponse(
    val data: List<SurveyModel>
)

data class GetSurveyResponse(
    val data: SurveyModel
)

data class SurveyModel(
    val id: Long,
    val title: String,
    val startDate: String,
    val endDate: String
)

fun toSurveyResponseList(surveys: List<SurveyModel>): List<SurveyResponse> {
    return surveys.map { survey ->
        SurveyResponse(
            id = survey.id,
            title = survey.title,
            startDate = survey.startDate,
            endDate = survey.endDate
        )
    }
}

fun toSurveyResponse(survey: SurveyModel): SurveyResponse {
    return SurveyResponse(
        id = survey.id,
        title = survey.title,
        startDate = survey.startDate,
        endDate = survey.endDate
    )
}

data class SurveyCreateRequest(
    val title: String,
    val startDate: String,
    val endDate: String
)

data class SurveyResponse(
    val id: Long,
    val title: String,
    val startDate: String,
    val endDate: String
)

data class Survey(
    val id: Long,
    val title: String,
    val startDate: String,
    val endDate: String
)
